src/BeOhmsWindow.cpp

```diff
@@ -1,30 +1,101 @@
 ///////////////////////////////////////////////////
 // BeOhmsWindow.cpp
-//
 // Copyright 1999  Kyle Crane
 //
 ///////////////////////////////////////////////////
 
-#ifndef _APPLICATION_H
-#include <Application.h>
-#endif
 
-#ifndef BEOHMSWINDOW_H
 #include "BeOhmsWindow.h"
-#endif
+#include <Application.h>
+#include <LayoutBuilder.h>
 
 
-/////////////////////////////////////////////////////
-// BeOhmsWindow -- Member Functions
-//
-BeOhmsWindow::BeOhmsWindow(BRect bounds)
-: BWindow(bounds, "BeOhms", B_TITLED_WINDOW, B_NOT_RESIZABLE | B_NOT_ZOOMABLE )
+
+BeOhmsWindow::BeOhmsWindow()
+: BWindow(BRect(),"BeOhms", B_TITLED_WINDOW, B_NOT_ZOOMABLE|B_NOT_RESIZABLE)
 {
-	// Size and setup the window
-	ResizeTo(285,190);
-	BRect myBounds(Bounds());
-	m_pMainView = new BeOhmsView(myBounds);
-	AddChild(m_pMainView);
+	// Size and center the window on screen
+	ResizeTo(320,240);
+	CenterOnScreen();
+	
+	back = new BView("back", B_WILL_DRAW);
+	
+	m_pSolveGroup = new BBox("solve_group");
+	m_pSolveGroup->SetLabel("Solve For");
+	
+	m_pOptVoltage = 
+		new BRadioButton("option_voltage", "Voltage", new BMessage(OPT_VOLTAGE));
+	m_pOptResistance = 
+		new BRadioButton("option_resistance", "Resistance", new BMessage(OPT_RESISTANCE));
+	m_pOptCurrent = 
+		new BRadioButton("option_current", "Current", new BMessage(OPT_CURRENT));
+	m_pOptCurrent->SetValue(1);
+	
+	
+	m_pTxtVoltage = new BTextControl("text_voltage", "", "", NULL);
+	m_pTxtResistance = new BTextControl("text_resistance", "", "", NULL);
+	
+	m_pTxtCurrent = new BTextControl("text_current", "", "", NULL);
+	m_pTxtCurrent->Hide();
+	
+	
+	m_pLblOutput = new BStringView("label_output", "Current ---->");
+	m_pTxtOutput = new BTextControl("text_output", "", "", NULL);
+	m_pTxtOutput->SetEnabled(false);
+	
+	m_pBtnCompute = 
+		new BButton("button_compute", "Compute", new BMessage(BTN_COMPUTE_PRESSED));  	
+	m_pBtnClear =
+		new BButton("button_clear", "Clear", new BMessage(BTN_CLEAR_PRESSED));
+	m_pBtnExit = 
+		new BButton("button_exit", "Exit", new BMessage(BTN_EXIT_PRESSED));
+			
+			
+	BStringView* pLblCopyright = new BStringView("copy_right", "Copyright 1999  Kyle Crane");
+		
+	
+	s_fBoxLayout = BLayoutBuilder::Group<>(B_HORIZONTAL)
+		.SetInsets(10)
+    	.AddGrid(B_USE_DEFAULT_SPACING, 0.0f)
+       	  	.AddGlue(0,0)
+    	  	.AddGlue(1,0)
```

Comment:
The position of the space layout item can be shifted in order to give desirable results when the window is resized.